runtime/lua/provider/python/health.lua

```diff
@@ -69,32 +69,40 @@ local function check_bin(bin)
 end
 
 -- Fetch the contents of a URL.
+---@return string error message. If success, returns an empty string ("").
 local function download(url)
   local has_curl = executable('curl')
-  if has_curl and vim.fn.system({ 'curl', '-V' }):find('Protocols:.*https') then
-    local out, rc = health.system({ 'curl', '-sL', url }, { stderr = true, ignore_error = true })
+  if
+    has_curl
+    and health._system({ 'curl', '-V' }, { ignore_error = true }):find('Protocols:.*https')
```

Comment:
Can we use `vim.system()` here directly? We don't need to ignore ENOENT errors here, so the only errors will be communicated via the exit code.